src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -220,19 +220,23 @@ internal virtual void Configure()
             // also assigns JsonPropertyInfo's JsonTypeInfo which causes SO if there are any
             // cycles in the object graph. For that reason properties cannot be added immediately.
             // This is a no-op for ReflectionJsonTypeInfo
-            LateAddProperties();
+            var propertyCache = LateAddProperties() ?? PropertyCache;
 
             DataExtensionProperty?.EnsureConfigured();
 
-            if (converter.ConverterStrategy == ConverterStrategy.Object && PropertyCache != null)
+            if (converter.ConverterStrategy == ConverterStrategy.Object && propertyCache != null)
             {
-                foreach (var jsonPropertyInfoKv in PropertyCache.List)
+                foreach (var jsonPropertyInfoKv in propertyCache.List)
```

Comment:
nit: should the declaring type's `NumberHandling` be passed to `EnsureConfigured()` as an argument?

Reply:
more likely `this` (declaring type JsonTypeInfo) perhaps because we don't really know if other info might be needed here. I'll refrain from changing it in this PR